### PR TITLE
associationlist and related roles have tediously long names

### LINKED ISSUE #1662: associationlist and related roles have tediously long names
`associationlist` and related roles have tediously long names. These look like ObjC naming conventions: `associationlist:item:value:` ;-)

> associationlist: A section containing associationlistitemkey and associationlistitemvalue elements.
> associationlistitemkey: A single key item in an association list.
> associationlistitemvalue: A single value item in an association list.

I don't have a particular preference other than "not what's there" but in the interest of forwarding the discussion, I'd suggest.

1. Remove `associationlist` entirely. I don't think there needs to be any distinction between `list` and `associationlist` in the Accessibility APIs. 
2. `listitemkey` and `listitemvalue` may be sufficient. It's possible `listitemvalue` could be an alias of `listitem`… I'm not certain, but the existence of a key may be all that's needed to distinguish between standard lists (~`html:l/html:ul`) and these (~`html:dl`)

### PR DESCRIPTION
**Pull request by jnurthen**
Fixes #1662

***

### :boom: Error: 504 Gateway Time-out :boom: ###

[PR Preview](https://github.com/tobie/pr-preview#pr-preview) failed to build. _(Last tried on May 11, 2022, 5:04 PM UTC)_.

<details>
<summary>More</summary>

PR Preview relies on a number of web services to run. There seems to be an issue with the following one:

:rotating_light: [Spec Generator](https://www.w3.org/2015/labs/) - Spec Generator is the web service used to build specs that rely on ReSpec.

:link: [Related URL](https://labs.w3.org/spec-generator/?type=respec&url=https%3A%2F%2Fraw.githubusercontent.com%2Fw3c%2Faria%2Fa7467cf6737b7ca99f5bbae8a19931a0a89511c2%2Findex.html%3FisPreview%3Dtrue)

```
<html><body><h1>504 Gateway Time-out</h1>
The server didn't respond in time.
</body></html>

```

_If you don't have enough information above to solve the error by yourself (or to understand to which web service the error is related to, if any), please [file an issue](https://github.com/tobie/pr-preview/issues/new?title=Error%20not%20surfaced%20properly&body=See%20w3c/aria%231740.)._
</details>

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -4588,6 +4410,7 @@ <h5>Note regarding the ARIA 1.3 <code>image</code> role.</h5>
 							<ul>
 								<li><code>&lt;ol&gt;</code> in [[HTML]]</li>
 								<li><code>&lt;ul&gt;</code> in [[HTML]]</li>
```
Comment (scottaohara) [inline index.html:4412]: worth adding in `<menu>` here as part of this pr?  or want me to create a separate PR just for this?

File: index.html [hunk 2]
```diff
@@ -4833,6 +4662,183 @@ <h5>Note regarding the ARIA 1.3 <code>image</code> role.</h5>
...
+				  <th class="role-parent-head" scope="row">Superclass Role:</th>
+				  <td class="role-parent"><rref>section</rref></td>
+				</tr>
+				<tr>
+				  <th class="role-children-head" scope="row">Subclass Roles:</th>
+				  <td class="role-children">Placeholder</td>
+				</tr>
+				<tr>
+				  <th class="role-base-head" scope="row">Base Concept:</th>
+				  <td class="role-base"><a href="https://www.w3.org/TR/html5/grouping-content.html#the-dt-element"><abbr title="Hypertext Markup Language">HTML</abbr> <code>dt</code></a></td>
```
Comment (scottaohara) [inline index.html:4701]:
```suggestion
				  <td class="role-base"><a href="https://html.spec.whatwg.org/multipage/grouping-content.html#the-dt-element"><abbr title="Hypertext Markup Language">HTML</abbr> <code>dt</code></a></td>
```

File: index.html [hunk 3]
```diff
@@ -4833,6 +4662,183 @@ <h5>Note regarding the ARIA 1.3 <code>image</code> role.</h5>
...
+			</table>
+		  </div>
+		  <div class="role" id="listitemvalue">
+			<rdef>listitemvalue</rdef>
+			<div class="role-description">
+			  <p>A single value item in a list.</p>
+	  
+			  <p>Author requirements for <a>elements</a> whose <a>role</a> is <code>listitemvalue</code>:</p>
+			  <ul>
+				<li>MUST be contained in an element whose <a>role</a> is <rref>list</rref>.</li>
```
Comment (scottaohara) [inline index.html:4759]: repeat of previous comment:
we should probably mention here that there should (must?) be no intervening listitem elements that interrupt the expected sibling relationship to the listitemkey/listitemvalue

File: index.html (hunk 1 above)
Comment (jnurthen) [reply index.html:4412]: I have never seen menu used anywhere so not sure I see the benefit of adding it.

Comment (scottaohara) [reply index.html:4412]: merely to be complete, since the spec lists 3 of the 4 related elements.

Comment (jnurthen) [reply index.html:4412]: I don't think it makes the spec any more understandable though as most devs would probably mistake the menu element for something useful

**Comment by pkra** [comment]
Overcome by events leading to #2095

**Pull request closed without merging**